sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -238,12 +294,61 @@ public static ErrorReport getLastSessionCrashReport() {
     }
 
     /**
+     * Provides information about any available crash report from the last session asynchronously, if it crashed.
+     *
+     * @param listener The listener that will receive crash in the last session.
+     */
+    public static void getLastSessionCrashReportAsync(LastCrashErrorReportListener listener) {
+        getInstance().getInstanceLastSessionCrashReportAsync(listener);
+    }
+
+    /**
+     * Implements {@link #hasCrashedInLastSession()} at instance level.
+     */
+    private synchronized boolean hasInstanceCrashedInLastSession() {
+        return mLastSessionCrashProcessingStatus != NOT_FOUND &&
+                mLastSessionCrashProcessingStatus != PREPARE_PROCESSING;
+    }
+
+    /**
      * Implements {@link #getLastSessionCrashReport()} at instance level.
      */
     private synchronized ErrorReport getInstanceLastSessionCrashReport() {
+        if (mLastSessionCrashProcessingStatus == PROCESSING) {
+            MobileCenterLog.debug(LOG_TAG, "Waiting for Crashes service to complete crash report for the last session.");
+            boolean failed = false;
+            try {
+                if (!mSemaphore.tryAcquire(GET_LAST_SESSION_CRASH_REPORT_TIMEOUT, TimeUnit.MILLISECONDS))
+                    failed = true;
+            } catch (InterruptedException e) {
+                failed = true;
+            }
+
+            if (failed)
+                MobileCenterLog.debug(LOG_TAG, "Could not get crash report for the last session.");
+        }
         return mLastSessionErrorReport;
     }
 
+    /**
+     * Implements {@link #getLastSessionCrashReportAsync(LastCrashErrorReportListener)} at instance level.
+     */
+    private synchronized void getInstanceLastSessionCrashReportAsync(LastCrashErrorReportListener listener) {
+        if (mLastSessionCrashProcessingStatus == PREPARE_PROCESSING ||
+                mLastSessionCrashProcessingStatus == PROCESSING) {
+            if (mLastCrashErrorReportListeners == null) {
+                mLastCrashErrorReportListeners = new ArrayList<>();
+            }
+            mLastCrashErrorReportListeners.add(listener);
+            MobileCenterLog.info(LOG_TAG, "Crashes for the last session have not been processed yet. The SDK will call listener when it completes process.");
+        } else if (mLastSessionCrashProcessingStatus == PROCESSED)
+            listener.onSuccess(mLastSessionErrorReport);
+        else if (mLastSessionCrashProcessingStatus == PROCESSING_FAILED)
+            listener.onFailure();
```

Comment:
We should not have a failure case for getSessionErrorReport, either return result or return null like before. The sync version has to return either a result or null and never throw an exception, so why would the async version introduce that?